### ISSUE TITLE
Is there a way to import data into Pug directly (without locals)?

### ISSUE THREAD

**Issue by wamoyo** [pugjs/pug#2758]
I'm using Pug in my static site, along with stylus, and some file syncing, it's nothing complex, but it helps me separate concerns further than plain HTML and CSS let me do. That being said, I have no server, I'm not running any app code in Node, just running some build scripts from my package.json and compiling my static site. 

I'd love to be able to separate some of my data from some HTML components. In my case I have speakers for an event series, like very many speakers over the years, and it would be nice to keep the speakers' bios and speech titles in a JSON file and load it into pug, where I can render it into the <speaker> component (not quite a full fledged web component, but it does the trick). 

The only way I've thought to do this is a javascript block inside a pug file, which seems at least a little abusive on my part.

Is there a better way that I missed somewhere in the documentation? Or am I requesting a feature here?

**Comment by Tiliavir** [comment]
Something like this?

```
  - var data = require("./data/data.json");
  
  each year in Object.keys(data)
```

**Comment by wamoyo** [comment]
Tried that. 

```terminal
  > 9|   - var speakers = require("../speakers.json")
    10|   each speaker in speakers
    11|     h2= speaker.name
    12|     p= speaker.talk

require is not a function
```

**Comment by Tiliavir** [comment]
Ah you are absolutely right - i forgot to send you half of the code ;-)

```
gulp.task("html:generatePages", function () {
    return gulp.src("./partials/pages/**/*.pug")
        .pipe($.data(function (file) { return { require: require }; };))
        .pipe($.pug())
        .pipe(gulp.dest(paths.dest));
});
```

I hope that helps - had to remove a lot from my gulp task, so I hope everything is still syntactically correct; but you will get the idea anyways ...

**Comment by wamoyo** [comment]
I don't use gulp. I use npm scripts. I see what you're doing there, passing the require function into pug. I kinda like that approach. Wonder if I can write it in npm scripts...

**Comment by wamoyo** [comment]
Hmmm... took 13 characters to solve this in npm scripts, haa.

Added `-O mydoc.json` to my pug line. I love when the tools I use have features I wasn't aware of, that are soooo valuable : )

**Comment by jsbimra** [comment]
@wamoyo can you share code link, how you were able to achieve this?

**Comment by wamoyo** [comment]
Of course! @jsbimra 

So, in my package.json I have an npm scripts section with a script that uses the --options flag from the pug cli

```
  "scripts": {
    "watch-pages": "pug -O data.js -w pages/ --out _static_site/"
}
```

Full list of options for pug cli here: https://github.com/pugjs/pug-cli#options (It's the -O or object or options flag.)

This file I think can be json, but I have a normal JavaScript file that exports a json object. Like so...

```
module.exports = {
  data: [1,2,3],
  NODE_ENV: process.env.NODE_ENV
}
```

I use the JS file so I can export the NODE_ENV path var.

Anywho, does this help?

**Comment by jsbimra** [comment]
I see, Thanks @Costa for reply, but I am trying to fetch data from JSON
file via making ajax call in JS file as it's take several secs to return me
the response but my pug template already render on page, can we make call
from pug mixin or something similar, but how? If you know example refering
to? I have also check on GitHub issues of using import or require to import
JSON data in pug template but it throws error require is not a function,
included it's CDN path in main ind X but still same error..

Basically I want to update or build pug template with asynchronous data
re-render part of template or after ajax request completed.

**Comment by wamoyo** [comment]
Ah, try Templatizer: https://github.com/henrikjoreteg/templatizer

**Comment by jsbimra** [comment]
Ok @costa I tried by following way for now, code here: doableyo.com/yes
very first post at page will check templatizer, thanks

**Comment by wamoyo** [comment]
Very cool site! I like the design : ) 

So, did you accomplish what you need with template strings in JS?

**Comment by jsbimra** [comment]
Yes @Costa, kind off for now, thanks :)
